### PR TITLE
Add opt to remove TypeInstance's revision, fix update mutation

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Add opt to remove TypeInstance's revision via dedicated gRPC method
- Store new revision in external storage before executing update cypher query
- Rollback stored revision in case of failed transaction

The Local Hub code was simplified thanks to handling the update logic directly in mutation resolver instead of `value` field resolver. However, the additionall logic (**rpc DeleteRevision**) is now required to be implemented on each storage backend.

## Testing

I added e2e tests for Local Hub to prove that the contract was not broken with the new approach.

## Related issue(s)

Fix https://github.com/capactio/capact/issues/661

### CODE REVIEW

File: internal/secret-storage-backend/server_test.go [hunk 1]
```diff
@@ -656,6 +657,146 @@ func TestHandler_OnDelete(t *testing.T) {
...
+				"cant-touch-this": {
+					"Music":        "hits me so hard",
+					"Makes me say": "Oh, my Lord",
+				},
+			}),
+			ExpectedProviderState: map[string]map[string]string{
+				path: {
+					"1": "key-one-value",
+				},
+				"cant-touch-this": {
```
Comment (mkuziemko) [inline internal/secret-storage-backend/server_test.go:713]: :+1:

File: internal/secret-storage-backend/server.go [hunk 2]
```diff
@@ -268,6 +269,40 @@ func (h *Handler) OnDelete(_ context.Context, request *pb.OnDeleteRequest) (*pb.
...
+// OnDeleteRevision handles TypeInstance's revision deletion by removing a secret entry in a given provider.
+// It checks whether a given TypeInstance is locked before doing such operation.
+func (h *Handler) OnDeleteRevision(ctx context.Context, request *pb.OnDeleteRevisionRequest) (*pb.OnDeleteRevisionResponse, error) {
+	if request == nil {
+		return nil, NilRequestInputError
+	}
+
+	provider, err := h.getProviderFromContext(request.Context)
+	if err != nil {
+		return nil, err
```
Comment (mkuziemko) [inline internal/secret-storage-backend/server.go:281]: should we use `errors.Wrap`?

**Review by mkuziemko — COMMENTED** [review]
I verified it locally and it works well :+1: small comments only

File: test/local-hub/external_storage_backend_test.go [hunk 3]
```diff
@@ -412,6 +415,120 @@ func TestExternalStorage(t *testing.T) {
...
+// TestDeleteTypeInstanceRevision can be run in parallel with other tests but
+// the subtests cannot be executed parallel.
+//
+// GRPC_SECRET_STORAGE_BACKEND_ADDR=50051 go test --tags=localhub -run=TestDeleteTypeInstanceRevision ./test/local-hub/...
+func TestDeleteTypeInstanceRevision(t *testing.T) {
+	// globally given
+	srvAddr := os.Getenv(storageBackendAddr)
+	if srvAddr == "" {
+		t.Skipf("skipping running test as the env %s is not provided", storageBackendAddr)
+	}
```
[the remark this answers is not in the document]
Comment (mszostok) [reply test/local-hub/external_storage_backend_test.go:427]: IMO it won't be worth it. As there are two options:
- don't pass any input args, but func needs to return multiple of them even though it's kinda not its responsibility
- extract and pass external args, this won't simplify the code

File: internal/secret-storage-backend/server.go (hunk 2 above)
Comment (mszostok) [reply internal/secret-storage-backend/server.go:281]: nope, in other places we also don't use it as the `getProviderFromContext` already returns proper grpc errors, and we don't want to override them

**Review by mkuziemko — APPROVED** [review]
LGTM :+1:

**Pull request merged**